Tests/xcodeprojTests/Objects/Configuration/XCBuildConfigurationTests.swift

```diff
@@ -18,6 +18,47 @@ final class XCBuildConfigurationTests: XCTestCase {
         XCTAssertEqual(XCBuildConfiguration.isa, "XCBuildConfiguration")
     }
 
+    func test_append_when_theSettingDoesntExist() {
+        // Given
+        let subject = XCBuildConfiguration(name: "Debug",
+                                           baseConfiguration: nil,
+                                           buildSettings: [:])
+
+        // When
+        subject.append(setting: "PRODUCT_NAME", value: "$(TARGET_NAME:c99extidentifier)")
+
+        // Then
+        XCTAssertEqual(subject.buildSettings["PRODUCT_NAME"] as? String, "$(inherited) $(TARGET_NAME:c99extidentifier)")
+    }
+
+    func test_append_when_theSettingExists() {
+        // Given
+        let subject = XCBuildConfiguration(name: "Debug",
+                                           baseConfiguration: nil,
+                                           buildSettings: ["OTHER_LDFLAGS": "flag1"])
+
+        // When
+        subject.append(setting: "OTHER_LDFLAGS", value: "flag2")
+
+        // Then
+        XCTAssertEqual(subject.buildSettings["OTHER_LDFLAGS"] as? String, "flag1 flag2")
```

Comment:
I noticed this failed locally for me

```swift
XCTAssertEqual failed: ("Optional("flag2 flag1")") is not equal to ("Optional("flag1 flag2")")
```

It's due to using a Set to reconstruct the setting. I think this test itself is correct as it is, appending settings should append the new setting to the end.